feat(curriculum): add moon orbit lab

zairahira opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->
To-Do:
- [ ] Link the animated prototype somewhere/ add to preview modal.


[Poject protoype here](https://github.com/freeCodeCamp/CurriculumExpansion/pull/432).
<!-- Feel free to add any additional description of changes below this line -->

zairahira committed:
feat: add project and tests

zairahira committed:
add more tests and stories

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
````diff
@@ -0,0 +1,262 @@
+Your `.earth` `div` element should have a background color.
+
+```js
+assert.exists(new __helpers.CSSHelp(document).getStyle('.earth')?.backgroundColor);
````
```suggestion
assert(new __helpers.CSSHelp(document).getStyle('.earth')?.backgroundColor);
```
This test is passing once the `.earth` selector is added to the stylesheet, even if `background-color` is not used.

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
````diff
@@ -0,0 +1,262 @@
+Your `.earth` element should have `border-radius` to make it look like a circle.
+
+```js
+assert.exists(new __helpers.CSSHelp(document).getStyle('.earth')?.borderRadius);
````
```suggestion
assert(new __helpers.CSSHelp(document).getStyle('.earth')?.borderRadius);
```

Dario-DC replied:
Also, to make it look like a circle maybe we should check that the value is set to 50%. What do you think?

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
````diff
@@ -0,0 +1,262 @@
+Your `.moon` `div` element should have a background color.
+
+```js
+assert.exists(new __helpers.CSSHelp(document).getStyle('.moon')?.backgroundColor);
````
```suggestion
assert(new __helpers.CSSHelp(document).getStyle('.moon')?.backgroundColor);
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
````diff
@@ -0,0 +1,262 @@
+Your `.moon` element should have `border-radius` to make it look like a circle.
+
+```js
+assert.exists(new __helpers.CSSHelp(document).getStyle('.moon')?.borderRadius);
````
```suggestion
assert(new __helpers.CSSHelp(document).getStyle('.moon')?.borderRadius);
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
````diff
@@ -0,0 +1,262 @@
+assert.exists(document.querySelector('div.space div.orbit div.moon'));
+```
+
+(combined with &&) The `moon` `div` should have a width and height of `30px`.
````
```suggestion
The `moon` `div` should have a width and height of `30px`.
```
I guess it was not supposed to be there

Dario-DC reviewed (changes requested):
I followed the user stories without doing anything else and the final result is a bit different from what expected. Personally I would add another couple of user stories so that the orbit will be centered onto the earth.

zairahira committed:
update dashed names and title

zairahira committed:
add boiler plate

zairahira commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
````diff
@@ -0,0 +1,262 @@
+## --seed-contents--
+
+```html
+
````
I thought the campers would add it themselves. I have provided one now.

zairahira committed:
fix: apply suggestions from code review

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

zairahira committed:
check border radius for circles

zairahira committed:
improve user stories

zairahira commented:
@Dario-DC, I have added some more user stories and tests for positioning of the elements.

Dario-DC reviewed (commented):
I was able to pass with this final result:
![kazam_im3gdzfl movie](https://github.com/user-attachments/assets/8be780fe-bcd4-46c7-8276-042a981328e0)

zairahira committed:
fix: enhance stories and tests

zairahira commented:
@Dario-DC, I have made the following adjustments to the stories and tests:

1. Checked order of `space`, `earth`, `moon` divs in html.
2. Added user stories and tests for aligning items in the `body`.
3. Added a user story and test for adding `transform: translateX(-50%);` to `.moon`

The new stories now would guide the campers in detail.

zairahira committed:
update naming convention

zairahira committed:
merge main

zairahira committed:
add block type

Dario-DC committed:
Merge branch 'main' of https://github.com/freeCodeCamp/freeCodeCamp into pr/zairahira/55670

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
````diff
@@ -0,0 +1,355 @@
+assert(new __helpers.CSSHelp(document).getStyle('.moon')?.position === 'absolute');
+```
+
+The `moon` `div` should position itself at the top of the `orbit` `div` and be centered horizontally.
````
I had some difficulties to understand that. Can we reword it in some way?

zairahira replied:
I have split this hint into two.

Dario-DC commented on client/i18n/locales/english/intro.json:
```diff
@@ -1713,6 +1713,12 @@
         "intro": ["", ""]
       },
+      "lab-moon-orbit": {
+        "title": "Build a Moon Orbit using CSS animation",
```
```suggestion
        "title": "Build a Moon Orbit",
```

Dario-DC commented on curriculum/challenges/_meta/lab-moon-orbit/meta.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "Build a Moon Orbit using CSS animation",
```
```suggestion
  "name": "Build a Moon Orbit",
```

zairahira replied:
@jdwilkin4 

Is it OK to use this naming convention?

jdwilkin4 replied:
Yeah, I think this name works 👍

Dario-DC reviewed (commented):
I left a couple of comments. It's looking good :+1: 
About the title: does using `Build X doing something` instead of `Build X` count as following the naming convention? If so, ignore my comments about the title.

zairahira committed:
add demoType

zairahira committed:
fix: add link in boilerplatte

zairahira committed:
fix: apply suggestions from code review

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

zairahira committed:
merge main

zairahira committed:
fix: update name

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

jdwilkin4 reviewed (commented):
Added one comment, but otherwise good

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
```diff
@@ -0,0 +1,362 @@
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
```
I wouldn't say that. Solar system refers to the Sun and the bodies orbiting around it.

What about this?
```md
In this lab, you will create a simple animation of the Moon's orbit around the Earth using HTML and CSS. The Earth will be at the center of the system, and the Moon will orbit around it.
```

jdwilkin4 replied:
Yeah, that fine. As long as we provide some so campers will not what they are getting into.

IMO, most of these labs should have descriptions because we want to make sure that campers understand what they are building.

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
```diff
@@ -0,0 +1,362 @@
+<head>
+    <meta charset="utf-8">
+    <title>Moon Orbit</title>
+    <link rel="stylesheet" href="styles.css">
```
```suggestion
```
Considering that we decided to remove the link from the seed code.

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
```diff
@@ -0,0 +1,362 @@
+   
+6. The `space` `div` should be centered on the page and have a width and height of `200px`.
+
+7. The `earth` `div` should use `absolute` positioning. Position the center of the `earth` `div` at the halfway point of its parent on both the vertical (top) and horizontal (left) axes. After that, shift the `earth` `div` back by half its own width and height, to center it within its parent, `space`.
```
```suggestion
7. The `earth` `div` should use `absolute` positioning. Position the center of the `earth` `div` at the halfway point of its parent on both the vertical (top) and horizontal (left) axes. After that, shift the `earth` `div` back by half its own width and height, to center it within its parent, `.space`.
```

Dario-DC reviewed (commented):
Left a few comments. I am a bit confused about the name. Have we decide to keep the "using CSS animation" part?

zairahira committed:
fix: apply suggestion from code review

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

zairahira committed:
add project description

zairahira committed:
fix: apply suggestion

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

zairahira committed:
add note to link stylesheet

zairahira commented:
@Dario-DC, the name has been updated in [this](https://github.com/freeCodeCamp/freeCodeCamp/pull/55670/commits/e0ba8dd9a01086c67db41978870ca66c77ebae04) commit. Seems fine or does it need to change?

Dario-DC commented:
> @Dario-DC, the name has been updated in [this](https://github.com/freeCodeCamp/freeCodeCamp/pull/55670/commits/e0ba8dd9a01086c67db41978870ca66c77ebae04) commit. Seems fine or does it need to change? Should I update it in `intro.json` as well?

Yes, I would update the `intro.json`.

zairahira committed:
refactor: update moon orbit title

zairahira commented:
Done!

moT01 commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
```diff
@@ -0,0 +1,365 @@
+
+4. Inside the `.orbit` `div`, there should be another `div` with the class `moon`.
+
+5. The `space` `div` should use `relative` positioning.
+   
+6. The `space` `div` should be centered on the page and have a width and height of `200px`.
```
These two can maybe be deleted? I don't see any tests for them, and I didn't add those properties to my project and it turned out as expected I think.

moT01 reviewed (commented):
This is looking good @zairahira 👍 

I found a user story that we can maybe delete?

I think [this comment I left on another lab applies here.](https://github.com/freeCodeCamp/freeCodeCamp/pull/55769#pullrequestreview-2255252287) I found myself just going down the list of tests and doing exactly what I was told. There wasn't a lot of room for creativity and I didn't have to think too hard. Maybe the same could be said for many of the projects. I'm wondering if we should try to make these more open-ended.

moT01 commented on curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md:
```diff
@@ -0,0 +1,365 @@
+
+4. Inside the `.orbit` `div`, there should be another `div` with the class `moon`.
+
+5. The `space` `div` should use `relative` positioning.
```
This one can maybe be deleted? I don't see a tests for it and I didn't add it to my project and it turned out as expected I think.

zairahira replied:
I have removed it.

zairahira committed:
remove relative positioning instruction

moT01 reviewed (approved):
I'm going to approve this so we can get it in. I don't particularly love it because of the reasons mentioned, but it was cool to see it come together. We have a lot of labs to make, so I think we can just get it in and maybe come back to it to make it more open-ended.

jdwilkin4 commented:
@moT01 

We have recently opened up the canary testing again on discord. 
So we will get some feedback from those testers on workshops and labs.
Plus, we will be able to test more and iterate from there once it is beta 👍
